### DISCUSSION TITLE
Some strange behaviors.,

### DISCUSSION THREAD

**Discussion by jamer508** [Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client#598] [Q&A]
I started using this Assistant tonight. It works great for controlling lights, the AC, smart TV volume and Volume of Google Home speakers. I can even broadcast to a few speakers. So why can I change the volume of a speaker that has a personalized name in the Home group, but not play music on it? I have all personal information permissions turned on any every other functions appears to work. 

I have read many articles about speaker groups and similar issues, I am just trying to understand how I can communicate to the google home speakers ( IE turning the volume up or down) but I cant do any of their other functions. 

Thanks guys.

**Comment by im-coder-lg** [comment]
Hi @jamer508,

I think we(especially @Melvin-Abraham) have seen such issues. I even noticed you commenting on a couple of those(got the notification).
I think it's primarily due to the Google Assistant SDK. Right now, we have really low compatibility for controlling home devices. I have none of them so I can't reproduce this...
It primarily could be the SDK, it's limited in its capabilities rather than the official Android app which uses the GMS services.(mentioned in the README.)
Maybe we could see this fixed in the future. As of now, I know only this. If you want broader opinions, you can ask @Melvin-Abraham or @sumeshir26, they know better.

**Comment by parano666** [comment]
First and foremost: THANKS. Have been googling in search of a solution like that periodically for years. 

About home smart products: works well here (turn on, turn off, etc, with custom names) but ROUTINE doesn't, even the basic one. 

Thanks again!

**Comment by im-coder-lg** [comment]
Oh, I think in the 2.0.0 Milestone Discussion, @Melvin-Abraham kinda mentioned(there or somewhere else) that 2.0.0 will also have Routines support. I wonder how the SDK will allow it tho...

**Reply by Melvin-Abraham** [reply]
I might have forgot to mention that the library doesn't provide access to user-created routines thus the routines support that is being planned to be implemented would be an unofficial one which would be separate from the official routines.

**Comment by Melvin-Abraham** [comment]
These capabilities are directly provided by the Google Assistant SDK, thus there's not entire surety about it's support. Support for media playback however may be added later on through extensions but may work on the desktop itself...

**Reply by parano666** [reply]
That would be awesome. Thanks again

**Comment by jamer508** [comment]
Thanks!
